src/Norway.js

```diff
@@ -0,0 +1,292 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import * as d3 from "d3";
+import { geoMercator, geoPath } from "d3-geo";
+import { feature } from "topojson-client";
+import counties from "./json/norway-counties.json";
+import municipalities from "./json/norway-municipalities.json";
+
+class Norway extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      countiesCollection: [],
+      municipalitiesCollection: [],
+      counties: [],
+      municipalities: []
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleCountyClick = this.handleCountyClick.bind(this);
+    this.showCountyTooltip = this.showCountyTooltip.bind(this);
+    this.hideCountyTooltip = this.hideCountyTooltip.bind(this);
+    this.handleMunicipalityClick = this.handleMunicipalityClick.bind(this);
+    this.showMunicipalityTooltip = this.showMunicipalityTooltip.bind(this);
+    this.hideMunicipalityTooltip = this.hideMunicipalityTooltip.bind(this);
```

Comment:
You can also do this as:
```
handleClick = () => {}
```

rather than binding in the constructor.

Reply:
Also, a lot of the code in this class doesn't really need to be part of the class, you could lift those up and just change their signatures to accept a "chart" or "svg" argument.